feat: fuse-overlayfs configuration (#2685)
codenvy/che-docs: modules/end-user-guide/pages/enabling-overlay-with-a-configmap.adoc

:_content-type: PROCEDURE
:description: You can enable fuse-overlayfs storage driver for Podman
:keywords: overlay, configmap, podman, storage, driver, fuse-overlayfs
:navtitle: Enabling fuse-overlayfs with a ConfigMap
:page-aliases:

[id="enabling-fuse-overlay-with-a-configmap"]
= Enabling fuse-overlayfs with a Configmap

> For Podman, the storage driver can be defined in the `~/.config/containers/storage.conf` file. Here are the default contents of the `/home/user/.config/containers/storage.conf` file in the UDI container:

Reviewer: ```suggestion
You can define the storage driver for Podman in the `~/.config/containers/storage.conf` file. Here are the default contents of the `/home/user/.config/containers/storage.conf` file in the UDI container:
```

.`storage.conf`
[source]
----
[storage]
driver = "vfs"
----

To use fuse-overlayfs, `storage.conf` can be set to the following:

.`storage.conf`
[source]
----
[storage]
driver = "overlay"

[storage.options.overlay]
mount_program="/usr/bin/fuse-overlayfs" <1>
----
<1> The absolute path to the `fuse-overlayfs` binary. The `/usr/bin/fuse-overlayfs` path is the default for the UDI.

> This can be done manually after starting a workspace. Another option is to build a new image based on the UDI with changes to `storage.conf` and use the new image for workspaces.

Reviewer: ```suggestion
You can do this manually after starting a workspace. Another option is to build a new image based on the UDI with changes to `storage.conf` and use the new image for workspaces.
```

> Otherwise, it is possible to update the `/home/user/.config/containers/storage.conf` for all workspaces in your {orch-namespace} by creating a ConfigMap that mounts the updated file. See xref:mounting-configmaps.adoc[].

Reviewer: ```suggestion
Otherwise, you can update the `/home/user/.config/containers/storage.conf` for all workspaces in your {orch-namespace} by creating a ConfigMap that mounts the updated file. See xref:mounting-configmaps.adoc[].
```

.Prerequisites

> * The administrator has enabled access to `/dev/fuse` by following xref:administration-guide:configuring-fuse.adoc[]

Reviewer: ```suggestion
* The administrator has enabled access to `/dev/fuse` by following xref:administration-guide:configuring-fuse.adoc[].
```

* A workspace with the required annotations are set by following xref:accessing-fuse.adoc[]

[NOTE]
====

> Since ConfigMaps mounted by following xref:mounting-configmaps.adoc[] mounts the ConfigMap's data to all workspaces, following this procedure will set the storage driver to fuse-overlayfs for all workspaces. Ensure that your workspaces contain the required annotations to use fuse-overlayfs by following xref:accessing-fuse.adoc[].

Reviewer: ```suggestion
Since ConfigMaps mounted by following xref:mounting-configmaps.adoc[this guide] mounts the ConfigMap's data to all workspaces, following this procedure will set the storage driver to fuse-overlayfs for all workspaces. Ensure that your workspaces contain the required annotations to use fuse-overlayfs by following xref:accessing-fuse.adoc[].
```
